src/inmanta/plugins.py

```diff
@@ -490,6 +427,10 @@ def get_annotation(arg: str) -> object:
 
             return arg_spec.annotations[arg]
 
+        # Make sure we have a return annotation even for implicit "null"
+        arg_spec.annotations["return"] = arg_spec.annotations.get("return") or "null"
```

Comment:
Why write it back to the arg_spec?

Reply:
I updated the explanation, is this more clear?